doc/guides/beyond-GSG.rst

```diff
@@ -0,0 +1,277 @@
+.. _beyond-gsg:
+
+Beyond the Getting Started Guide
+################################
+
+The :ref:`getting_started` gives a straight-forward path to set up
+your Linux, macOS, or Windows enviromnent for Zephyr development. In
+this document, we delve deeper into Zephyr development setup
+issues and alternatives.
+
+.. _python-pip:
+
+Python and pip
+**************
+
+Python 3 and its package manager, pip\ [#pip]_, are used extensively by Zephyr
+to install and run scripts required to compile and run Zephyr
+applications, set up and maintain the Zephyr development environment,
+and build project documentation.
+
+Depending on your operating system, you may need to provide the
+``--user`` flag to the ``pip3`` command when installing new packages. This is
+documented throughout the instructions.
+See `Installing Packages`_ in the Python Packaging User Guide for more
+information about pip\ [#pip]_, including `information on -\\-user`_.
+
+- On Linux, make sure ``~/.local/bin`` is at the front of your :envvar:`PATH`
+  :ref:`environment variable <env_vars>`, or programs installed with ``--user``
+  won't be found. Installing with ``--user`` avoids conflicts between pip
+  and the system package manager, and is the default on Debian-based
+  distributions.
+
+- On macOS, `Homebrew disables -\\-user`_.
+
+- On Windows, see the `Installing Packages`_ information on ``--user`` if you
+  require using this option.
+
+On all operating systems, pip's ``-U`` flag installs or updates the package if the
+package is already installed locally but a more recent version is available. It
+is good practice to use this flag if the latest version of a package is
+required.  (Check the :zephyr_file:`scripts/requirements.txt` file to
+see if a specific Python package version is expected.)
+
+Advanced Setup and tool chain alternatives
+******************************************
+
+Here are some alternative instructions for more advanced platform setup
```

Comment:
Need to decide what we want to have here - repetition of the GSG, alternative setups, or troubleshooting. Currently, it's a bit of a mix.